File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/MilitaryEntity.java
package dev.ragnarok.fenrir.db.model.entity;

import androidx.annotation.Keep;

@Keep
public class MilitaryEntity {

    private String unit;

    private int unitId;

    private int countryId;

    private int from;

    private int until;

    public int getUntil() {
        return until;
    }

    public MilitaryEntity setUntil(int until) {
        this.until = until;
        return this;
    }

    public int getFrom() {
        return from;
    }

    public MilitaryEntity setFrom(int from) {
        this.from = from;
        return this;
    }

    public int getCountryId() {
        return countryId;
    }

    public MilitaryEntity setCountryId(int countryId) {
        this.countryId = countryId;
        return this;
    }

    public int getUnitId() {
        return unitId;
    }

    public MilitaryEntity setUnitId(int unitId) {
        this.unitId = unitId;
        return this;
    }

    public String getUnit() {
        return unit;
    }

    public MilitaryEntity setUnit(String unit) {
        this.unit = unit;
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/MilitaryEntity.kt
package dev.ragnarok.fenrir.db.model.entity

import androidx.annotation.Keep
import kotlinx.serialization.Serializable

@Keep
@Serializable
class MilitaryEntity {
    var unit: String? = null
        private set
    var unitId = 0
        private set
    var countryId = 0
        private set
    var from = 0
        private set
    var until = 0
        private set

    fun setUntil(until: Int): MilitaryEntity {
        this.until = until
        return this
    }

    fun setFrom(from: Int): MilitaryEntity {
        this.from = from
        return this
    }

    fun setCountryId(countryId: Int): MilitaryEntity {
        this.countryId = countryId
        return this
    }

    fun setUnitId(unitId: Int): MilitaryEntity {
        this.unitId = unitId
        return this
    }

    fun setUnit(unit: String?): MilitaryEntity {
        this.unit = unit
        return this
    }
}